### PR TITLE
Replace generic reagent stamina buffs with buffs specific to reagents

### PR DESCRIPTION
**Pull request by Kyle2143**
## About the PR 
Previously, stamina buffs/debuffs from reagents all used one of two keys, "consumable_good" and "consumable_bad". This meant that if you had two different reagents in your bloodstream that changed your stamina, then the last reagent that was added was the stamina buff that was applied. This was a minor bug.

Also, it prevented you from using two good chems, in order to get more stamina regen than either of them. The opposite is true as well. 

The other main change made here is a performance change. Previously, each add or removal of a reagent stamina buff/debuff used this long conditional that uses reflection:
``istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_regen")
``
And I changed them to: 
```
if(ismob(holder?.my_atom))
	var/mob/M = holder.my_atom
```
Because mobs are the only type that have stamina.

## Why's this needed? 
This is sort of fixing a bug, but it also opens up some new possibilities in chem-cocktails for messing with stamina. It was my understanding that this was a change that was always meant to happen and that "consumable_good" was just temporary to quickly give reagents more stamina flavor without a ton of effort.

## Changelog

```
(u)Kyle:
(+)Stamina buffs and debuffs of different reagents now stack with eachother. You can use two stamina buff chems to add their effects together.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
whoops, didn't push the whole branch.

**Comment by mybluecorners** [comment]
would still like these wrapped into status effects. otherwise looks fine

**Comment by mordent-goonstation** [comment]
> would still like these wrapped into status effects. otherwise looks fine

Agreed, but this is still a step in the right direction.

File: code/modules/chemistry/Reagents-Drugs.dm [hunk 1]
```diff
@@ -472,13 +474,13 @@ datum
 
 			on_add()
 				if(istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_regen"))
-					remove_buff = holder.my_atom:add_stam_mod_regen("consumable_good", 1)
+					remove_buff = holder.my_atom:add_stam_mod_regen("r_nicotine", 1)
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-Drugs.dm:477]: Any reason you didn't do the `if (ismob(holder?.my_atom))` etc. stuff here in addition? There's a few others you didn't touch either. I think we should be consistent with how we check/apply them, at least.

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 2]
```diff
@@ -905,13 +907,13 @@ datum
 
 			on_add()
 				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_max"))
-					remove_buff = holder.my_atom:add_stam_mod_max("consumable_bad", -20)
+					remove_buff = holder.my_atom:add_stam_mod_max("r_pancuronium", -30)
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-PoisonEtc.dm:910]: Looks like you changed this from `-20` to `-30`.

File: code/modules/chemistry/Reagents-Medical.dm [hunk 3]
```diff
@@ -428,14 +432,16 @@ datum
 			stun_resist = 27
 
 			on_add()
-				if(istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_regen"))
-					remove_buff = holder.my_atom:add_stam_mod_regen("consumable_good", 2)
+				if(ismob(holder?.my_atom))
+					var/mob/M = holder.my_atom
+					remove_buff = M.add_stam_mod_regen("r_synaptizine", 4)
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-Medical.dm:437]: Looks like you buffed `2` to `4`?

File: code/modules/chemistry/Reagents-Medical.dm [hunk 4]
```diff
@@ -605,14 +611,16 @@ datum
 			var/remove_buff = FALSE
 
 			on_add()
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_regen"))
-					remove_buff = holder.my_atom:add_stam_mod_regen("consumable_good", 1)
+				if(ismob(holder?.my_atom))
+					var/mob/M = holder.my_atom
+					remove_buff = M.add_stam_mod_regen("r_smelling_salt", 5)
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-Medical.dm:616]: Looks like you buffed `1` to `5`?

File: code/modules/chemistry/Reagents-PoisonEtc.dm (hunk 2 above)
Comment (Kyle2143) [reply code/modules/chemistry/Reagents-PoisonEtc.dm:910]: Yeah, that was intentional. 20 seemed too low for panc

**Comment by Kyle2143** [comment]
Oh yeah, I suppose I forgot to say. I did change a couple stamina values that I noticed seemed a bit off imo; as mordent pointed out.

**Pull request merged**